app/code/Magento/Elasticsearch/Elasticsearch5/Model/Adapter/FieldMapper/Product/FieldProvider/FieldType/Resolver/CompositeResolver.php

```diff
@@ -33,6 +33,10 @@ public function __construct(array $items)
             }
         }
         $this->items = $items;
+        unset($this->items['default']);
```

Comment:
Please consider to use sortOrder attribute of array elements that can be specified in DI
This hardcoded 'default' key looks unreliable for me

Reply:
done